### ISSUE TITLE
Why is the cwd_loss is 0 when I use your self-distill commond?

### ISSUE THREAD

**Issue by hjg12345** [huawei-noah/Efficient-Computing#95]
I follow your work in Gold-yolo, and use the commond to train a teacher model ,then use the self-distill commond to train student model. But the cwd_loss is 0, as is shown below!
![image](https://github.com/huawei-noah/Efficient-Computing/assets/127812522/14551e6e-6652-4feb-a065-e0ff1ddb2a1f)

**Comment by lose4578** [comment]
Can you give me more details, like your model config and data yaml. Does this affect the mIoU of the model.

**Comment by hjg12345** [comment]
![image](https://github.com/huawei-noah/Efficient-Computing/assets/127812522/87b355a1-a887-43ee-99a6-381b9307ef99)
the model is gold-yolo-s, dataset is coco2017.  I have known why the cwd_loss is 0 (beacuse without using --distill_feat), but I still want to know why the cls_loss is negativt? And the self-distill performance seems to be lower.

**Comment by kaka-Cao** [comment]
> ![image](https://user-images.githubusercontent.com/127812522/277271029-87b355a1-a887-43ee-99a6-381b9307ef99.png) the model is gold-yolo-s, dataset is coco2017. I have known why the cwd_loss is 0 (beacuse without using --distill_feat), but I still want to know why the cls_loss is negativt? And the self-distill performance seems to be lower.

i have the same situation. the self-distill is lower than origin.

**Comment by kaka-Cao** [comment]
> Can you give me more details, like your model config and data yaml. Does this affect the mIoU of the model.

if using both the --distill and --distill_feat are better than just the --distill

**Comment by lose4578** [comment]
> ![image](https://user-images.githubusercontent.com/127812522/277271029-87b355a1-a887-43ee-99a6-381b9307ef99.png) the model is gold-yolo-s, dataset is coco2017. I have known why the cwd_loss is 0 (beacuse without using --distill_feat), but I still want to know why the cls_loss is negativt? And the self-distill performance seems to be lower.

Please give specific training commands and model config.

**Comment by lose4578** [comment]
This may be due to the teacher's checkpoints not being aligned with the model, resulting in the error label generated by the teacher. You can set `strict=True` in here  [load_state_dict](https://github.com/huawei-noah/Efficient-Computing/blob/master/Detection/Gold-YOLO/yolov6/utils/checkpoint.py#L19) to check your teacher model checkpoint  are correct.

**Comment by hjg12345** [comment]
What is the epoch number when training for Teacher and Student model?

**Comment by lose4578** [comment]
300 and 300
